pyccel/ast/numpyext.py

```diff
@@ -786,22 +786,22 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
+    _shape = ()
+    _rank  = 0
 
     def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
-            sh = list(sh)
-            del sh[self.dim]
-            self._shape = tuple(sh)
-        else:
-            self._shape = ()
-        self._rank = len(self._shape)
         super().__init__(arg, dim)
+        if isinstance(arg.dtype, NativeBool):
+            arg = PythonFloat(PythonInt(arg))
+        elif isinstance(arg.dtype, NativeComplex):
+            arg = PythonAbs(arg)
```

Comment:
Is this needed? Does fortran's `norm` not do this?

Reply:
Fortran `norm` doesn't do this.

Reply:
ok. I wonder if this is the cause of the imprecision? It is not great to do a square root and a square:
```
n = sqrt(sqrt(a**2 + b**2)**2 + ...)
```
Is there any way to translate:
```
norm(a)
```
to:
```
norm([*a.real, *a.imag])
```
? 
Ideally only in the fortran printer?

Reply:
and in case of arrays ?

Reply:
Hence the unpacking `*a.real`, I'm not sure how you can write this in pyccel though, I'll leave this up to you to investigate